libnd4j/include/ops/declarable/platform/armcompute/armcomputeUtils.cpp

```diff
@@ -0,0 +1,291 @@
+  }
+  // https://arm-software.github.io/ComputeLibrary/v20.05/_dimensions_8h_source.xhtml
+  // note: underhood it is stored as std::array<T, num_max_dimensions> _id;
+  // TensorShape is derived from Dimensions<uint32_t>
+  // as well as Strides : public Dimensions<uint32_t>
+  Arm_TensorShape shape;
+  Arm_Strides strides;
+  shape.set_num_dimensions(rank);
+  strides.set_num_dimensions(rank);
+  size_t element_size = arm_compute::data_size_from_type(dType);
+  for (int i = 0, j = rank - 1; i < rank; i++, j--) {
+    shape[i] = static_cast<uint32_t>(bases[j]);
+    strides[i] = static_cast<uint32_t>(arrStrides[j]) * element_size;
+  }
+  // fill the rest unused with 1
+  for (int i = rank; i < arm_compute::MAX_DIMS; i++) {
+    shape[i] = 1;
+  }
+  size_t total_size;
+  size_t size_ind = rank - 1;
+  total_size = shape[size_ind] * strides[size_ind];
+
+  Arm_TensorInfo info;
+  info.init(shape, numChannels, dType, strides, 0, total_size);
+  info.set_data_layout(layout);
+
+  return info;
+}
+
+Arm_Tensor getArmTensor(const NDArray& arr, arm_compute::DataLayout layout) {
+  // - Ownership of the backing memory is not transferred to the tensor itself.
+  // - The tensor mustn't be memory managed.
+  // - Padding requirements should be accounted by the client code.
+  // In other words, if padding is required by the tensor after the function
+  // configuration step, then the imported backing memory should account for it.
+  // Padding can be checked through the TensorInfo::padding() interface.
+
+  // Import existing pointer as backing memory
+  auto info = getArmTensorInfo(arr, layout);
+  Arm_Tensor tensor;
+  tensor.allocator()->init(info);
+  void* buff = (void*)arr.buffer();
+  tensor.allocator()->import_memory(buff);
+  return tensor;
+}
+
+void copyFromTensor(const Arm_Tensor& inTensor, NDArray& output) {
+    //only for C order
+    //only for C order
+    if (output.ordering() != 'c') return;
+    auto shapeInfo = output.shapeInfo();
+    auto bases = &(shapeInfo[1]);
+    Nd4jLong rank = shapeInfo[0];
+    auto strides = output.stridesOf();
+    int width = bases[rank - 1];
+    uint8_t* outputBuffer = (uint8_t*)output.buffer(); 
+    size_t offset = 0;
+    arm_compute::Window window;
+    arm_compute::Iterator tensor_it(&inTensor, window);
+
+    int element_size = inTensor.info()->element_size();
+    window.use_tensor_dimensions(inTensor.info()->tensor_shape(), /* first_dimension =*/arm_compute::Window::DimY);
+
+    if (output.ews() == 1) {
+        auto copySize = width * element_size;
+        auto dest = outputBuffer;
+        arm_compute::execute_window_loop(window, [&](const arm_compute::Coordinates& id)
+            {
+                auto src = tensor_it.ptr(); 
+                memcpy(dest, src, copySize);
+                dest += copySize;
+            },
+            tensor_it);
+    }
+    else {
+        Nd4jLong coords[MAX_RANK] = {};
+        if(strides[rank-1]!=1){ 
```

Comment:
Same as above. It's possible to have permuted array with last stride being equal to 1.